packages/ra-core/src/dataProvider/withLifecycleCallbacks.ts

```diff
@@ -67,13 +67,13 @@ import {
  *   [
  *     {
  *       resource: "posts",
- *       afterRead: async (data, dataProvider) => {
+ *       afterRead: async (data, {dataProvider, resource}) => {
```

Comment:
This is a breaking change. You can add the same feature in a backward compatible way, by adding a third parameter instead.

Reply:
changed